src/main/java/com/iexec/worker/tee/TeeWorkflowConfigurationService.java

```diff
@@ -0,0 +1,73 @@
+package com.iexec.worker.tee;
+
+import com.iexec.sms.api.SmsClient;
+import com.iexec.sms.api.SmsClientProvider;
+import com.iexec.sms.api.TeeWorkflowConfiguration;
+import com.iexec.worker.docker.DockerService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manages the {@link TeeWorkflowConfiguration}, providing an easy way to get a configuration for a task
+ * and avoiding the need to create a new {@link TeeWorkflowConfiguration} instance each time.
+ */
+@Slf4j
+@Service
+public class TeeWorkflowConfigurationService {
+    private final SmsClientProvider smsClientProvider;
+    private final DockerService dockerService;
+
+    // TODO: purge this map when a task has been completed
+    private final Map<String, TeeWorkflowConfiguration> configurationForTask = new HashMap<>();
+
+    public TeeWorkflowConfigurationService(SmsClientProvider smsClientProvider,
+                                           DockerService dockerService) {
+        this.smsClientProvider = smsClientProvider;
+        this.dockerService = dockerService;
+    }
+
+    public TeeWorkflowConfiguration getOrCreateTeeWorkflowConfiguration(String chainTaskId) {
+        return configurationForTask.computeIfAbsent(chainTaskId, this::buildTeeWorkflowConfiguration);
+    }
+
+    TeeWorkflowConfiguration buildTeeWorkflowConfiguration(String chainTaskId) {
+        // SMS client should already have been created once before.
+        // If it couldn't be created, then the task would have been aborted.
+        // So the following won't throw an exception.
+        final SmsClient smsClient = smsClientProvider.getOrCreateSmsClientForTask(chainTaskId);
+
+        final TeeWorkflowConfiguration config = smsClient.getTeeWorkflowConfiguration();
+        log.info("Received tee workflow configuration [config:{}]", config);
+        if (config == null) {
+            throw new TeeWorkflowConfigurationCreationException(
+                    "Missing tee workflow configuration [chainTaskId:" + chainTaskId +"]");
+        }
+
+        final String preComputeImage = config.getPreComputeImage();
+        final String postComputeImage = config.getPostComputeImage();
+
+        if (!dockerService.getClient(preComputeImage)
```

Comment:
I don't think force pull is required here.
If image is not present, pull it, else keep going

Reply:
I disagree. In case an image is rebuilt and uses the same tag as before - bad practice, but we sometimes have this issue -, then we'd like to pull.
BTW, I'm pretty sure Docker will only pull missing layers, not the complete image. So if we already have all the layers, it will simply be a fast check, right?

Reply:
FYI:
"Docker will only pull missing layers" true
"it will simply be a fast check" it depends how the image is built (example, if an apt update/upgrade is made in first layers, next layers will be all different)

Reply:
Right, please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/50f398c3b157f77bd60123395e673a5b71814d56.